lib/adapters/class-vip-enterprise-search.php

```diff
@@ -197,6 +255,168 @@ public function filter__vip_search_post_taxonomies_allow_list( $taxonomies ) {
 		return $this->get_restricted_taxonomies() ?: $taxonomies;
 	}
 
+	/**
+	 * A callback for the wp_rest_search_handlers filter. Includes the
+	 * search-suggestions REST API handler in the list of allowed handlers if
+	 * search suggestions are made available over REST.
+	 *
+	 * @param \WP_REST_Search_Handler[] $search_handlers List of search handlers to use in the controller.
+	 * @return \WP_REST_Search_Handler[] Updated list of handlers.
+	 */
+	public function filter__wp_rest_search_handlers( $search_handlers ) {
+		if ( $this->is_show_search_suggestions_in_rest_enabled() ) {
+			$search_handlers[] = new Post_Suggestion_Search_Handler( $this );
+		}
+
+		return $search_handlers;
+	}
+
+	/**
+	 * Suggest posts that match the given search term.
+	 *
+	 * @param string $search Search string.
+	 * @param array  $args   {
+	 *     Optional. An array of arguments.
+	 *
+	 *     @type string[] $subtypes Limit suggestions to this subset of all post
+	 *                              types that support search suggestions.
+	 *     @type int      $page     Page of results.
+	 *     @type int      $per_page Results per page. Default 10.
+	 *     @type int[]    $include  Search within these post IDs.
+	 *     @type int[]    $exclude  Exclude these post IDs from results.
+	 *     @type string[] $status   Post statuses to search. Default 'publish'.
+	 * }
+	 * @return int[] Post IDs in this page of results and total number of results.
+	 */
+	public function query_post_suggestions( string $search, array $args = [] ): array {
+		$out = [ [], 0 ];
+
+		if ( ! $search && ! $this->get_allow_empty_search() ) {
+			return $out;
+		}
+
+		$args = wp_parse_args(
+			$args,
+			[
+				'subtypes' => [],
+				'page'     => 1,
+				'per_page' => 10,
+				'exclude'  => [], // phpcs:ignore WordPressVIPMinimum.Performance.WPQueryParams.PostNotIn_exclude
+				'include'  => [],
+				'status'   => [ 'publish' ],
+			],
```

Comment:
🤔  Should we add a hook in case it might be needed?

Reply:
A hook does make sense here, I think, since it doesn't have to reference the ES provider.